autoload/nnn.vim

```diff
@@ -342,7 +345,19 @@ function! nnn#pick(...) abort
     let l:default_opts = { 'edit': 'edit' }
     let l:opts = extend(l:default_opts, get(a:, 2, {}))
     let s:temp_file = tempname()
-    let l:cmd = g:nnn#command.' -p '.shellescape(s:temp_file).' '.(l:directory != '' ? shellescape(l:directory): '')
+    if g:nnn#session ==# 'none'
+        let l:sess_cfg = ' '
+    elseif g:nnn#session ==# 'global'
+        let l:sess_cfg = ' -S '
+    elseif g:nnn#session ==# 'local'
+        let l:sess_cfg = ' -S -s '.s:local_ses.' '
+        let session_file = s:nnn_conf_dir.'/sessions/'.s:local_ses
+        if !(exists('g:nnn_ses_autocmd'))
+            execute 'autocmd VimLeavePre * call delete(fnameescape("'.session_file.'"))'
+            let g:nnn_ses_autocmd = 1
```

Comment:
This flag is sort of annoying. I plan on using an `augroup` instead. @mcchrish Do you have any thoughts?

Reply:
Please feel free to change.

Reply:
Is it strictly required to store the session in the config directory? I thought it's possible to make use of any session file that is stored anywhere.

Only problem I think is when a rare occasion that vim/nvim crashed and was not able to run VimLeavePre and such, not able to clean up the temp session file.

Reply:
Though the path is fixed, it's well documented in `nnn`.

To counter this problem, please use a constant prefix for the session file and document it so users can identify them easily.

Reply:
Maybe something like `vim-*`? Should I still use `tempname()` in that case?

Reply:
> Only problem I think is when a rare occasion that vim/nvim crashed and was not able to run VimLeavePre and such, not able to clean up the temp session file.

I cannot think of any solution to this.

Reply:
nnn-vim-*

Yes, please create tmpfile using timestamp, if possible.

Reply:
Ah, I understand what you mean. Well, in that case a prefix with `vim-...` would be sufficient it seems.

Reply:
Yes, that too works.